### ISSUE TITLE
Material UI resets size adjustments applied to icons

### ISSUE THREAD

**Issue by leozdgao** [DanBrooker/file-icons#296]
After I update the file-icons the `.tag` icon looks too big in the tree view. And I already restart Atom after I update this package.

![image](https://cloud.githubusercontent.com/assets/8381024/13552405/2d8e975e-e39c-11e5-9ab2-f4a85e6eb471.png)

The icon on the tab looks good:

![image](https://cloud.githubusercontent.com/assets/8381024/13552414/80fe31d8-e39c-11e5-8eb7-924ec7fef17b.png)

I try to customize this icon with new stylesheet like this:

```css
.filetype-icon-tag:before {
  transform: scale(.5);
}
```

Still not working...

**Comment by Alhadis** [comment]
What UI theme are you using?

**Comment by leozdgao** [comment]
I'm using (atom-material-ui)[https://atom.io/themes/atom-material-ui]  v1.2.0

**Comment by Alhadis** [comment]
Ugh, I notice the theme is overriding size adjustments that File-Icons applies to individual glyphs:

<img src="https://cloud.githubusercontent.com/assets/2346707/13552447/d9bb7806-e3b7-11e5-82fd-86ad05694561.png" width="308" />

It's affecting other non-square icons too (TeX and Lisp, for example):

<img src="https://cloud.githubusercontent.com/assets/2346707/13552456/a24ac3f8-e3b8-11e5-9f7e-10ef96a5ffac.png" width="332" />

In other words, this extends beyond the Riot icon specifically.

**Comment by Alhadis** [comment]
@DanBrooker Having Atom Material reset the icon sizes is somewhat problematic, because it means some are too small (note the Laravel and Rust icons), while others are too big.

There're two possible approaches we can take:

1. **Override Atom Material's icon sizes**, which would make everything else look too big to everybody using the theme. It would guarantee a bit more consistency with the icon-size adjustments, however.
2. **Edit the sizes of each icon being used in this theme's fonts** to ensure they fit snugly into a 16×16 canvas. This would be a lot of work, and would basically copying the existing icons into a single font. This sounds stupid, but it would actually ensure harmonious sizes and alignments across icons, resolving #260 as well as this.
3. **Have ad hoc styling corrections applied for Material users** which seems to be the easy and logical thing to do, but it's only solving half the problem.

**Comment by leozdgao** [comment]
For now, I use this custom stylesheet to mitigate the issue:

```css
.icon-file-text[data-name$=".tag"]:before,
.icon-file-binary[data-name$=".tag"]:before {
  font-size: 7px !important;
}
```

**Comment by Alhadis** [comment]
I'll cut a patch release to address this particular issue in the short-term, because this is actually bad enough to warrant it.

Mark my words, though: something has to be done about the inconsistent sizes and offsets. Mfizz in particular seems to warrant it most - look at how misaligned the Scala and SQL icons are:

<img src="https://cloud.githubusercontent.com/assets/2346707/13552830/fd60d98c-e3c5-11e5-8ae9-bd49065fb9b4.png" width="600" />

**Comment by DanBrooker** [comment]
The best solution (the least hacky) would be an entire custom font with nicely aligned and sized icons, but that's a large amount of work and a whole project on it's own

I don't know if this would help but maybe the file icon service is way to avoid this problem  (See #229 https://github.com/atom/tree-view#api ) which is the offical way to provide icons and might get around the conflict between themes somehow (maybe?).

**Comment by Alhadis** [comment]
It is, but honestly, I'd be more than willing to do the heavy lifting. I even started working [on a way to automate this sort of thing](https://github.com/Alhadis/IconForge), although it's still in development. Basically if I finished that, it'd allow recompilation of the icon font using a folder of 16×16-sized SVG files. It'd make auditing and modification a hell of a lot easier, and we wouldn't be at the mercy of every font author's different spacing and size considerations.

**Comment by DanBrooker** [comment]
@Alhadis this might be useful https://github.com/ryanoasis/nerd-fonts

**Comment by Alhadis** [comment]
Hrm, no, not quite what I had in mind, haha. =) Basically, that repo seems to be patching existing fonts, whereas this would be generating one from scratch. 

Don't worry, I'll nut something out! Just in the middle of other projects right now, heh.

**Comment by Alhadis** [comment]
Also, some of those icons still appear to be unevenly-sized, so we're still stuck with the issue of inconsistent sizing...

**Comment by shernshiou** [comment]
I can confirm with this issue.

![image](https://cloud.githubusercontent.com/assets/64935/14594332/e8593750-0567-11e6-98be-374ffaf62b02.png)

**Comment by Alhadis** [comment]
Ugh, good lord. Fixing that right now...

**Comment by Alhadis** [comment]
What the hell... this thing isn't even fitted to the canvas correctly! :|

<img width="600" alt="" src="https://cloud.githubusercontent.com/assets/2346707/14594432/f583ce88-0579-11e6-8b6e-b98d215be22d.png">

**Comment by Alhadis** [comment]
Okay guys, I've normalised the widths of every icon that didn't have a square canvas. The issue should be resolved for now... although the other icon fonts really need the normalisation I spoke about.

This will be pushed out in a patch release shortly.

**Comment by shernshiou** [comment]
Updated to 1.7.4. Still facing the same problem.

![image](https://cloud.githubusercontent.com/assets/64935/14597302/6b5004a2-057e-11e6-9fec-6afac3a180d6.png)

**Comment by Alhadis** [comment]
@shernshiou Did you restart Atom after installing?

**Comment by shernshiou** [comment]
@Alhadis ahh! that solved the problem. Thanks!

**Comment by Alhadis** [comment]
Good to hear!

**Comment by edwinksl** [comment]
Looks good on Atom 1.8.0-beta1 too.
